graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java

```diff
@@ -646,4 +684,29 @@ public FieldTypeException(Throwable e) {
             super(e);
         }
     }
+
+
+    /**
+     * Extracts the last stream id from the filter string passed as part of the elasticsearch query. This is used later
+     * to pass to possibly existing message decorators for stream-specific configurations.
```

Comment:
This extracts only the last stream id from the filter? That breaks when you do `stream_id:foo OR stream_id:bar`, no? It will then only search in the index set of the last stream?